### DISCUSSION TITLE
[错误报告]: 刮削文件权限问题, UMASK 000 创建出的文件 Others 用户不具备权限

### DISCUSSION THREAD

**Discussion by Matzoh** [jxxghp/MoviePilot#809] [讨论]
### 确认

- [X] 我的版本是最新版本，我的版本号与 [version](https://github.com/jxxghp/MoviePilot/releases/latest) 相同。
- [X] 我已经 [issue](https://github.com/jxxghp/MoviePilot/issues) 中搜索过，确认我的问题没有被提出过。
- [X] 我已经 [Telegram频道](https://t.me/moviepilot_channel) 中搜索过，确认我的问题没有被提出过。
- [X] 我已经修改标题，将标题中的 描述 替换为我遇到的问题。

### 当前程序版本

1.3.1

### 问题类型

主程序运行问题

### 问题描述

创建出的文件和文件夹权限问题, 我通过smb共享的media, 供windows等电脑用, 需要 Others 用户具有读写权限

UMASK设置的 000
PUID和 PGID 都是 0

期望logo.png及刮削后的视频目录权限也是: -rw-rw-rw-
emby现在删除提示无权限, emby是其他服务器的, windows的

### 发生问题时系统日志和配置文件

```bash
刮削前文件夹权限:
drwxrwxrwx 1 1000 users   98 Oct 12 17:45 铃芽之旅.Suzume.no.Tojimari.2022.UHD.BluRay.2160p.x265.10bit.HDR.2Audio.mUHD-FRDS/

刮削前文件权限:
-rw-rw-rw- 2 1000 users 9.8G Oct 12 14:41 Suzume.no.Tojimari.2022.UHD.BluRay.2160p.x265.10bit.HDR.2Audio.mUHD-FRDS.mkv

刮削后出的文件夹权限:
drwxr-xr-x 1 root   root   251 Oct 12 17:45 铃芽之旅\ (2022)/

刮削后的文件权限:
-rw-r--r-- 1 root root   35K Oct 12 17:45 logo.png
-rw-rw-rw- 2 1000 users 9.8G Oct 12 14:41 铃芽之旅\ (2022)\ -\ 2160p.mkv

期望logo.png权限也是: -rw-rw-rw-
```

**Comment by dlovew** [comment]
这个跟mp没关系吧，你安装puid和guid与文件下载器一致就行了

**Comment by Matzoh** [comment]
> 这个跟mp没关系吧，你安装puid和guid与文件下载器一致就行了

硬链文件权限是没问题的, 但刮削出来的png/nfo等文件权限是很低的, 比如logo.png, 变成rw-r--r--了

**Comment by dlovew** [comment]
> > 这个跟mp没关系吧，你安装puid和guid与文件下载器一致就行了
> 
> 刮削出来新的文件权限是更低的, 比如logo.png, 变成rw-r--r--了

所以你统一mp和下载器的权限就行了呗，还有emby权限

**Comment by Matzoh** [comment]
> > > 这个跟mp没关系吧，你安装puid和guid与文件下载器一致就行了
> > 
> > 
> > 刮削出来新的文件权限是更低的, 比如logo.png, 变成rw-r--r--了
> 
> 所以你统一mp和下载器的权限就行了呗，还有emby权限

好几台服务器, 而且有的还是windows什么的, 能把权限open了就不用再挨个排查这个了

**Comment by AceCandy** [comment]
所以这个问题解决了吗，我用飞牛，qb的用户和组是qBittorrent、qBittorrent，emby的用户和组是EmbyServer、EmbyServer，mp给了0、0、000，刮削图片是-rw-rw----+权限，导致emby出图片的时候没有权限了，手动将图片权限设为777之后就能读取到  这个有什么办法处理么

**Comment by lie5860** [comment]
确实还是存在这个问题。在升级到v2。我需要把emby的uid gid字段设为0 才能够查看封面。虽然也能解决我的问题就是了。封面无法访问 在群晖上显示是root用户创建的。尝试在mp上设置pid和gid用当前用户的id无效。

**Comment by tobarcn** [comment]
这个问题不但影响刮削,还影响复制方式的文件移动

复制之后的文件,所有权限全部丢失,变成了000权限

这是qb下载后的原始文件权限
`-rwxrwxrwx+ 1 root     root           10868999912 Mar  4 15:30 'House of Wolves 2016 BluRay 1080p x264 DTS-HD MA 5.1 2Audios-MTeam.mkv'`

这是经过mp复制之后的文件权限
`----------  1 root     root           10868999912 Mar  4 15:30 'House of Wolves 2016 BluRay 1080p x264 DTS-HD MA 5.1 2Audios-MTeam.mkv'
`

但是很奇怪的是,我在复制过程中看过权限,当时是对的,但是复制完成之后权限就全丢了，这个是复制过程中的，看大小能看出是未完成状态
`-rwxrwxrwx+ 1 root     root           922746880 Mar  5 18:10 'House of Wolves 2016 BluRay 1080p x264 DTS-HD MA 5.1 2Audios-MTeam.mkv'
`

这个是权限变化过程
![image](https://github.com/user-attachments/assets/df7d0278-20ff-4e4f-88c9-88d1cdd2ec39)
